packages/web3-eth-accounts/src/index.js

```diff
@@ -201,30 +201,32 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
                     delete transaction.hardfork;
                 }
             }
-
+            
             if (privateKey.startsWith('0x')) {
                 privateKey = privateKey.substring(2);
             }
-
-            var ethTx = new Transaction(transaction, transactionOptions);
-
-            ethTx.sign(Buffer.from(privateKey, 'hex'));
-
-            var validationResult = ethTx.validate(true);
-
-            if (validationResult !== '') {
-                throw new Error('Signer Error: ' + validationResult);
+            
+            var ethTx = Transaction.fromTxData(transaction, transactionOptions);
+            var signedTx = ethTx.sign(Buffer.from(privateKey, 'hex'));
+            var validationResult = signedTx.validate(true);
```

Comment:
might make sense to rename this to `validationErrors` since the return values look like [this](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/baseTransaction.ts)